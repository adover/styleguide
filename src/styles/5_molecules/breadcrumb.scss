@import '../1_settings/spacing';
@import '../1_settings/svg';
@import '../2_tools/mixins/apply-spacing';
@import '../2_tools/mixins/center';
@import '../2_tools/mixins/list-reset';
@import '../4_atoms/link';
@import '../4_atoms/list';
@import '../4_atoms/paragraph';
/// @group molecules

$breadcrumb-separator-size: $spacing-3;
$breadcrumb-separator-color: 'regent-gray';

/// TODO
/// @name breadcrumb
@mixin breadcrumb() {
	@include list-init($reset: true, $margin-bottom: false);
	@include section-margin($size: default);

	font-size: map-deep-get($config, sizes, paragraph, small);
}

/// TODO
/// @name breadcrumb-separator
@mixin breadcrumb-separator() {
	background-image: svg($breadcrumb-separator-color, $path: 'chevron-right');
	background-size: 100%;
	height: $breadcrumb-separator-size;
	width: $breadcrumb-separator-size;
}

/// TODO
/// @name breadcrumb-item
@mixin breadcrumb-item($current: false) {
	@if not $current {
		@include list-item-inline($left-margin: 0);

		a {
			@include link-init();
		}
	}
}
