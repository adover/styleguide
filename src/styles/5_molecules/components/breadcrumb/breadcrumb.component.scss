@import '../../../4_atoms/list.scss';
@import '../../../1_settings/_config.scss';
@import '../../../1_settings/_spacing.scss';
@import '../../../2_tools/mixins/apply-spacing.scss';

// TODO: to move this to the css level once we kill off reskin
:host {
	ul {
		@include list-init($reset: true, $inline: true);
		@include apply-spacing($axis: horizontal);

		li {
			align-items: center;
			display: inline-flex;
			font-size: map-deep-get($config, sizes, paragraph, small);

			cdx-svg-icon {
				&[name='chevron-right'] {
					margin: 0 $spacing-1;
					position: relative;
					top: 1px;
				}
			}
		}
	}
}
