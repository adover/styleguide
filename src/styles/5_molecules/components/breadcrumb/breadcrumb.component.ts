import { Component, Input, OnInit } from '@angular/core';
import { Router } from '@angular/router';

@Component({
	selector: 'cdx-breadcrumb',
	template: `
		<ul *ngIf="items.length">
			<li>
				<a routerLink="/" class="_noajax" aria-label="Home">
					<cdx-svg-icon name="home" size="small" fill="dark"></cdx-svg-icon>
				</a>
				<cdx-svg-icon name="chevron-right" fill="dark" size="extra-small"></cdx-svg-icon>
			</li>
			<li *ngFor="let item of items; let last = last">
				<ng-container *ngIf="item.routeUrl">
					<a [routerLink]="item.routeUrl" [queryParams]="item.queryParams">{{ item.text | titlecase }}</a>
				</ng-container>
				<ng-container *ngIf="!item.routeUrl">
					<span>{{ item.text | titlecase }}</span>
				</ng-container>
				<div class="separator"></div>
				<cdx-svg-icon name="chevron-right" fill="dark" size="extra-small" *ngIf="!last"></cdx-svg-icon>
			</li>
		</ul>
	`,
	styleUrls: ['./breadcrumb.component.scss'],
})
export class BreadcrumbComponent implements OnInit {
	@Input() items: any[] = [];

	constructor(private router: Router) {}

	ngOnInit(): void {}

	navigate = (link: string) => {
		this.router.navigateByUrl(link);
	};
}
