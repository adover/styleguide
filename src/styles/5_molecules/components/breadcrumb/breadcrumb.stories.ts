import { RouterTestingModule } from '@angular/router/testing';
import { storiesOf } from '@storybook/angular';
import { BreadcrumbComponent } from './breadcrumb.component';
import { SvgDefinitionsComponent, SvgIconComponent } from '@woolworthsnz/styleguide';

storiesOf('Molecules| Breadcrumb', module).add('default', () => ({
	moduleMetadata: {
		declarations: [SvgDefinitionsComponent, SvgIconComponent, BreadcrumbComponent],
		imports: [RouterTestingModule],
	},
	template: `
    <cdx-svg-definitions></cdx-svg-definitions>
    <cdx-breadcrumb></cdx-breadcrumb>
  `,
}));
