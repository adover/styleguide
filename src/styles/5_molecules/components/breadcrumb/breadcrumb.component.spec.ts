import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { SvgIconComponent } from '@woolworthsnz/styleguide';
import { BreadcrumbComponent } from './breadcrumb.component';

describe('BreadcrumbComponent', () => {
	let component: BreadcrumbComponent;
	let fixture: ComponentFixture<BreadcrumbComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [BreadcrumbComponent, SvgIconComponent],
			imports: [RouterTestingModule, BrowserAnimationsModule],
		}).compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(BreadcrumbComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		component.items = [
			{ text: 'Groceries', routeUrl: '/', queryParams: {} },
			{
				text: 'Foo',
				routeUrl: 'shop/browse/foo',
				queryParams: {
					search: 'milk',
					page: 1,
					size: 24,
					sort: undefined,
					dasFilter: 'foo;1;Foo;false',
					filters: null,
				},
			},
			{
				text: 'Bar',
				routeUrl: 'shop/browse/foo/bar',
				queryParams: {
					search: 'milk',
					page: 1,
					size: 24,
					sort: undefined,
					dasFilter: 'bar;1;Bar;false',
					filters: null,
				},
			},
			{
				text: 'Baz',
				routeUrl: 'shop/browse/foo/bar/baz',
				queryParams: {
					search: 'milk',
					page: 1,
					size: 24,
					sort: undefined,
					dasFilter: 'baz;1;Baz;false',
					filters: null,
				},
			},
		];
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

	it('should match snapshot with item not having route url', () => {
		component.items = [
			{ text: 'Groceries', routeUrl: '/', queryParams: {} },
			{
				text: 'Foo',
				routeUrl: 'shop/browse/foo',
				queryParams: {
					search: 'milk',
					page: 1,
					size: 24,
					sort: undefined,
					dasFilter: 'foo;1;Foo;false',
					filters: null,
				},
			},
			{
				text: 'Bar',
				routeUrl: 'shop/browse/foo/bar',
				queryParams: {
					search: 'milk',
					page: 1,
					size: 24,
					sort: undefined,
					dasFilter: 'bar;1;Bar;false',
					filters: null,
				},
			},
			{
				text: 'Baz',
				routeUrl: '',
				queryParams: {},
			},
		];
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});
});
