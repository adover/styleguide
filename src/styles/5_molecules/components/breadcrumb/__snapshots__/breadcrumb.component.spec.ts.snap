// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent should match snapshot 1`] = `
<cdx-breadcrumb
  __ngContext__={[Function LRootView]}
  items={[Function Array]}
  navigate={[Function Function]}
  router={[Function Router]}
>
  <ul>
    <li>
      <a
        aria-label="Home"
        class="_noajax"
        href="/"
        routerlink="/"
      >
        <cdx-svg-icon
          display="block"
          fill="dark"
          name="home"
          size="small"
        >
          <svg
            version="1.1"
            viewbox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlink:href="#home"
            />
          </svg>
        </cdx-svg-icon>
      </a>
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
    </li>
    <li>
      <a
        href="/"
      >
        Groceries
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      <a
        href="/shop/browse/foo?search=milk&page=1&size=24&dasFilter=foo;1;Foo;false"
      >
        Foo
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      <a
        href="/shop/browse/foo/bar?search=milk&page=1&size=24&dasFilter=bar;1;Bar;false"
      >
        Bar
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      <a
        href="/shop/browse/foo/bar/baz?search=milk&page=1&size=24&dasFilter=baz;1;Baz;false"
      >
        Baz
      </a>
      
      
      
      <div
        class="separator"
      />
      
    </li>
    
  </ul>
</cdx-breadcrumb>
`;

exports[`BreadcrumbComponent should match snapshot with item not having route url 1`] = `
<cdx-breadcrumb
  __ngContext__={[Function LRootView]}
  items={[Function Array]}
  navigate={[Function Function]}
  router={[Function Router]}
>
  <ul>
    <li>
      <a
        aria-label="Home"
        class="_noajax"
        href="/"
        routerlink="/"
      >
        <cdx-svg-icon
          display="block"
          fill="dark"
          name="home"
          size="small"
        >
          <svg
            version="1.1"
            viewbox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlink:href="#home"
            />
          </svg>
        </cdx-svg-icon>
      </a>
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
    </li>
    <li>
      <a
        href="/"
      >
        Groceries
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      <a
        href="/shop/browse/foo?search=milk&page=1&size=24&dasFilter=foo;1;Foo;false"
      >
        Foo
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      <a
        href="/shop/browse/foo/bar?search=milk&page=1&size=24&dasFilter=bar;1;Bar;false"
      >
        Bar
      </a>
      
      
      
      <div
        class="separator"
      />
      <cdx-svg-icon
        display="block"
        fill="dark"
        name="chevron-right"
        size="extra-small"
      >
        <svg
          version="1.1"
          viewbox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <use
            xlink:href="#chevron-right"
          />
        </svg>
      </cdx-svg-icon>
      
    </li>
    <li>
      
      <span>
        Baz
      </span>
      
      
      <div
        class="separator"
      />
      
    </li>
    
  </ul>
</cdx-breadcrumb>
`;
